### PR TITLE
[#3290] Fix Git conflicted files error of #3307

### PR DESCRIPTION
**Pull request by phuongdm**
### Pull Request Checklist: 
- [ ] Positive Test Case Written by Dev

- [ ] Automated Testing

- [ ] Sufficient User and Developer Documentation

- [ ] Passing Jenkins Build

- [ ] Peer Code review and approval

### Positive Test Case
1. Please follow this link to test: https://docs.google.com/document/d/1_8ClNiIBVSEI3jY4UJKOWrV6uwKilKug-69LoVizezk/edit

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -18,10 +18,13 @@ __pycache__/
 /nginx/.htpasswd
 /nginx/htpasswd
 /nginx/Dockerfile
+/nginx/Dockerfile-nginx
```
Comment (cuahsimarko) [inline .gitignore]: I'd like to see all the Dockerfiles in one folder (the project root or a dockerfiles folder), however if this is too much work to fix leave it as-is

When working with the old mstealy build it was a nightmare tracking down all the files

File: conf_irods/partial_build.sh [hunk 2]
```diff
@@ -7,52 +7,84 @@ echo "echo rods | iinit" | $RUN_ON_USER
```
Comment (cuahsimarko) [inline conf_irods/partial_build.sh]: eventually will this script go away? was trying to reduce it's size not grow it. we can do that in a separate PR if it's too much work

File: config/hydroshare-config.yaml [hunk 3]
```diff
@@ -33,16 +33,16 @@ IRODS_LOCAL_MOUNT: /irods
 ### These give Nginx direct access to iRODS vaults
 ### All ROOTs must be children of DOCKER_MAP ###
 IRODS_DOCKER_MAP: /irods
-IRODS_DATA_ROOT: /irods/data
-IRODS_USER_ROOT: /irods/user
+IRODS_DATA_ROOT: /irods/data/iRODS/Vault/home/wwwHydroProxy
+IRODS_USER_ROOT: /irods/user/iRODS/Vault/home/localHydroProxy
 IRODS_CACHE_ROOT: /irods/cache
```
Comment (cuahsimarko) [inline config/hydroshare-config.yaml]: do these overlap with settings.py and local_settings.py?

File: config/hydroshare-config.yaml [hunk 4]
```diff
@@ -33,16 +33,16 @@ IRODS_LOCAL_MOUNT: /irods
...
 
 ### Nginx worker process must be able to read iRODS files.
 ### These used to build the nginx docker container.
 SENDFILE_IRODS_USER: irods
 SENDFILE_IRODS_GROUP: irods
-SENDFILE_IRODS_USER_ID: 20000
-SENDFILE_IRODS_GROUP_ID: 20000
+SENDFILE_IRODS_USER_ID: 998
+SENDFILE_IRODS_GROUP_ID: 998
```
Comment (cuahsimarko) [inline config/hydroshare-config.yaml]: @hyi made an earlier comment about hardwiring account ids I believe. is this ok here?

File: hsctl [hunk 5]
```diff
@@ -4,15 +4,20 @@
 # HydroShare Control Script
 # Author: Michael Stealey <michael.j.stealey@gmail.com>
 
+machine="`uname -s`"
+case "$machine" in
+  Linux*)  export SED_EXT=''   ;;
+  Darwin*) export SED_EXT='""' ;;
+  *)       export SED_EXT=''   ;;
+esac
+
```
Comment (cuahsimarko) [inline hsctl]: what about Windows? the CUAHSI standard right now is Min GW (the Git for Windows shell)

Comment (cuahsimarko) [reply hsctl]: If it's too hard to do "all 3" in a single process, we can have two entirely separate setups for windows vs everything else

File: .gitignore (hunk 1 above)
Comment (cuahsimarko) [reply .gitignore]: actually leave this alone. I see you are using the relative path context in the docker-compose yml

File: local-dev.yml [hunk 6]
```diff
@@ -113,10 +111,29 @@ services:
...
+      - "temp_vol:/hs_tmp"
+      - "share_vol:/irods/cache"
+      - "idata_vault_vol:/irods/data"
+      - "iuser_vault_vol:/irods/user"
   hydroshare:
     build:
       context: .
-      dockerfile: Dockerfile-hydroshare
+      dockerfile: Dockerfile
     container_name: hydroshare
```
Comment (cuahsimarko) [inline local-dev.yml]: ideally every dockerfile will differentiate itself and there will be none named just 'Dockerfile' in the entire project

File: nginx/Dockerfile-nginx.template [hunk 7]
```diff
@@ -0,0 +1,11 @@
+FROM nginx:1.11
```
Comment (cuahsimarko) [inline nginx/Dockerfile-nginx.template]: I really want to get away from mstealy's use of "dockerfile-templates" and shell scripts that modify them. Let's just have actual dockerfiles everywhere and none of the dynamic templating of dockerfiles. this is a pretty strong requirement unless it's a huge time investment. please refactor or let's discuss.

File: nginx/Dockerfile.template [hunk 8]
```diff
@@ -7,10 +7,11 @@ RUN cp /tmp/config-files/hs-nginx.conf /etc/nginx/conf.d/default.conf \
     && cp /tmp/.htpasswd /etc/nginx/.htpasswd \
     && cp /tmp/docker-entrypoint.sh /docker-entrypoint.sh \
     && groupadd -g SENDFILE_IRODS_GROUP_ID SENDFILE_IRODS_GROUP \
-    && useradd --system --shell /bin/bash -u SENDFILE_IRODS_USER_ID -g SENDFILE_IRODS_GROUP_ID SENDFILE_IRODS_USER 
+    && echo 'SENDFILE_IRODS_USER:x:SENDFILE_IRODS_USER_ID:SENDFILE_IRODS_GROUP_ID::/home/SENDFILE_IRODS_USER:/bin/bash' >> /etc/passwd
```
Comment (cuahsimarko) [inline nginx/Dockerfile.template]: this seems messy but I suppose as long as it works

File: scripts/templates/init-defaultworker.template [hunk 9]
```diff
@@ -1,7 +1,9 @@
 #!/usr/bin/env bash
 
 cd /hydroshare
-usermod -u HS_SERVICE_UID hydro-service
-groupmod -g HS_SERVICE_GID storage-hydro
```
Comment (cuahsimarko) [inline scripts/templates/init-defaultworker.template]: are there not cleaner user management routines?

File: conf_irods/partial_build.sh (hunk 2 above)
Comment (phuongdm) [reply conf_irods/partial_build.sh]: We will remove this script on other PR, this PR's purpose is only make life simpler for new developer

File: config/hydroshare-config.yaml (hunk 3 above)
Comment (phuongdm) [reply config/hydroshare-config.yaml]: yes, that's settings on data.local.org and users.local.org. If dev knows how to config/modify that means he/she no needs this PR

File: config/hydroshare-config.yaml (hunk 4 above)
Comment (phuongdm) [reply config/hydroshare-config.yaml]: That's irods UID inside data.local.org and users.local.org. If we change it, means we use other iRODS settings ===> no needs this PR

File: hsctl (hunk 5 above)
Comment (phuongdm) [reply hsctl]: Linux is Linux
Mac is Darwin
Windows is * 
=))

File: local-dev.yml (hunk 6 above)
Comment (phuongdm) [reply local-dev.yml]: So what will happen if in future we have to use other hs_docer_base image, modify all Dockerfile???

File: nginx/Dockerfile-nginx.template (hunk 7 above)
Comment (phuongdm) [reply nginx/Dockerfile-nginx.template]: Yes, in this PR, we are able to fix it to be Dockerfile-nginx, no need customizable

File: nginx/Dockerfile.template (hunk 8 above)
Comment (phuongdm) [reply nginx/Dockerfile.template]: useradd won't work with UID > 65535 (on CUAHSI dev-hs-* most of domain UID is greater than 65535) :( so modify /etc/passwd directly is the solution

File: scripts/templates/init-defaultworker.template (hunk 9 above)
Comment (phuongdm) [reply scripts/templates/init-defaultworker.template]: Same above

**Comment by cuahsimarko** [comment]
Ok good responses, Phuong. I will test build Monday with Mahesh

**Comment by cuahsimarko** [comment]
Tested working

**Comment by cuahsimarko** [comment]
Please merge if/when Jenkins done
